Quizzler-iOS13/SceneDelegate.swift

```diff
@@ -13,11 +13,34 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     var window: UIWindow?
 
 
-    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+    func scene(
+        _ scene: UIScene,
+        willConnectTo session: UISceneSession,
+        options connectionOptions: UIScene.ConnectionOptions
+    ) {
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
-        guard let _ = (scene as? UIWindowScene) else { return }
+        
+        
+        // 1. Capture the scene if there is one into a variable
+        guard let windowScene = (scene as? UIWindowScene) else { return }
+        
+        // 2. Create a new UIWindow using the windowScene constructor which takes in a window scene
+        let window = UIWindow(windowScene: windowScene)
+        
+        // 3. Create the view hierarchie programmatically
+        let viewController = ViewController()
+        let navigationController = UINavigationController(rootViewController: viewController)
```

Comment:
_Side note, not a scope of this PR_

For a single page application it is a valid solution, at the same time the more pages and the more conditions to when open which page you'll have — the more difficult it be to handle navigation in this way.

There are different techniques how to improve it, one way is to use Coordinator pattern:
https://medium.com/@maysam.shahsavari/a-comprehensive-guide-to-coordinator-pattern-in-swift-7e7647ecc525
https://medium.com/@dkw5877/flow-coordinators-333ed64f3dd
https://www.hackingwithswift.com/articles/71/how-to-use-the-coordinator-pattern-in-ios-apps

Reply:
Thank you! I see 👍🏽